jaq-parse/src/token.rs

```diff
@@ -180,7 +180,7 @@ pub fn tree(
     let trees = tree.map_with_span(|t, span| (t, span)).repeated().collect();
     let paren = trees.clone().delimited_by(just('('), just(')'));
     let brack = trees.clone().delimited_by(just('['), just(']'));
-    let brace = trees.clone().delimited_by(just('{'), just('}'));
+    let brace = trees.delimited_by(just('{'), just('}'));
```

Comment:
I do not like this change, because it breaks the symmetry with the lines above. Perhaps make `trees` a closure and call `trees()` instead of `trees.clone()`?

Reply:
I found a nice solution to this in ee4d75404136705817821a7b9c0cfca895d2707b.
It shortens the overall code, while restoring symmetry.
Yes, in principle it involves one more clone, but these clones are not performance-critical.